go/libraries/doltcore/schema/typeinfo/bit_test.go

```diff
@@ -0,0 +1,253 @@
+			374562394,
+			0,
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf(`%v %v`, test.typ.String(), test.input), func(t *testing.T) {
+			output, err := test.typ.ConvertNomsValueToValue(test.input)
+			if test.expectedErr {
+				assert.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, test.output, output)
+			}
+		})
+	}
+}
+
+func TestBitConvertValueToNomsValue(t *testing.T) {
+	tests := []struct {
+		typ         *bitType
+		input       interface{}
+		output      types.Uint
+		expectedErr bool
+	}{
+		{
+			generateBitType(t, 1),
+			true,
+			1,
+			false,
+		},
+		{
+			generateBitType(t, 5),
+			int16(25),
+			25,
+			false,
+		},
+		{
+			generateBitType(t, 64),
+			uint64(287946293486),
+			287946293486,
+			false,
+		},
+		{
+			generateBitType(t, 3),
+			float32(78.3),
+			0,
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf(`%v %v`, test.typ.String(), test.input), func(t *testing.T) {
+			output, err := test.typ.ConvertValueToNomsValue(test.input)
+			if !test.expectedErr {
+				require.NoError(t, err)
+				assert.Equal(t, test.output, output)
+			} else {
+				assert.Error(t, err)
+			}
+		})
+	}
+}
+
+func TestBitFormatValue(t *testing.T) {
+	tests := []struct {
+		typ         *bitType
+		input       types.Uint
+		output      string
+		expectedErr bool
+	}{
+		{
+			generateBitType(t, 1),
+			0,
+			"0",
+			false,
+		},
+		{
+			generateBitType(t, 1),
+			1,
+			"1",
+			false,
+		},
+		{
+			generateBitType(t, 11),
+			250,
+			"250",
+			false,
+		},
+		{
+			generateBitType(t, 64),
+			math.MaxUint64,
+			strconv.FormatUint(math.MaxUint64, 10),
+			false,
+		},
+		{
+			generateBitType(t, 1),
+			7,
+			"",
+			true,
+		},
+		{
+			generateBitType(t, 10),
+			374562394,
+			"",
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf(`%v %v`, test.typ.String(), test.input), func(t *testing.T) {
+			output, err := test.typ.FormatValue(test.input)
+			if test.expectedErr {
+				assert.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, test.output, *output)
+			}
+		})
+	}
+}
+
+func TestBitParseValue(t *testing.T) {
+	tests := []struct {
+		typ         *bitType
+		input       string
+		output      types.Uint
+		expectedErr bool
+	}{
+		{
+			generateBitType(t, 1),
+			"0",
+			0,
+			false,
+		},
+		{
+			generateBitType(t, 5),
+			"25",
+			25,
+			false,
+		},
+		{
+			generateBitType(t, 64),
+			"287946293486",
+			287946293486,
+			false,
+		},
+		{
+			generateBitType(t, 3),
+			"78.3",
+			0,
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf(`%v %v`, test.typ.String(), test.input), func(t *testing.T) {
+			output, err := test.typ.ParseValue(&test.input)
+			if !test.expectedErr {
+				require.NoError(t, err)
+				assert.Equal(t, test.output, output)
+			} else {
+				assert.Error(t, err)
+			}
+		})
+	}
+}
+
+func generateBitTypes(t *testing.T, numOfTypes uint16) []TypeInfo {
```

Comment:
Might want to add a note to these methods that they are used in the other test file. Either that or move them there. It's a little odd to share code between test files like this.